coops-ui/src/views/AnnualReport.vue

```diff
@@ -408,25 +426,51 @@ export default {
         )
       }
 
-      axios.post(this.corpNum + '/filings', filingData).then(res => {
-        let payRequestId: string = res.data.filing.header.paymentToken
-        // TODO: use return URL / param to display Dashboard with paid filing expanded?
-        let returnURL = window.location.origin + '/AnnualReport?pay_id=' + payRequestId
-        let authStub: string = this.authURL
-        if (!(authStub.endsWith('/'))) {
-          authStub = authStub + '/'
-        }
-        let payURL = authStub + 'makepayment/' + payRequestId + '/' + encodeURIComponent(returnURL)
-        // TODO: need to check if pay UI is reachable, else display modal dialog
-        window.location.href = payURL
-      }).catch(error => {
-        console.error('submit() error =', error)
-        if (error.response && error.response.status === 402) {
-          this.paymentErrorDialog = true
-        } else {
-          this.saveErrorDialog = true
-        }
-      })
+      if (this.filingId) {
+        // we have a filing id, so we are updating an existing filing
+        const url = this.corpNum + '/filings/' + this.filingId
+        axios.put(url, filingData).then(res => {
+          let payRequestId: string = res.data.filing.header.paymentToken
+          // TODO: use return URL / param to display Dashboard with paid filing expanded?
+          let returnURL = window.location.origin + '/AnnualReport?pay_id=' + payRequestId
+          let authStub: string = this.authURL
+          if (!(authStub.endsWith('/'))) {
+            authStub = authStub + '/'
+          }
+          let payURL = authStub + 'makepayment/' + payRequestId + '/' + encodeURIComponent(returnURL)
+          // TODO: need to check if pay UI is reachable, else display modal dialog
+          window.location.href = payURL
+        }).catch(error => {
+          console.error('submit() error =', error)
+          if (error.response && error.response.status === 402) {
```

Comment:
We should consider the value of using the http-status-codes library, but I think at some point we're probably going to want a consolidation of our API-calling code.

Reply:
Great idea!
Done.